user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfigBuilder.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * User Operator Configuration Builder class
+ */
+public class UserOperatorConfigBuilder {
+
+    private final Map<String, String> map;
+
+    protected  UserOperatorConfigBuilder() {
+        this.map = new HashMap<>();
+    }
+
+    protected  UserOperatorConfigBuilder(UserOperatorConfig config) {
```

Comment:
The builder looks good. But I wonder if you can maybe more automate it ... 
* You know the config keys, so can you maybe just iterate through the options instead of listing them one by one?
* As this is expected to be used by tests only basically, do we need the specific `.with...()` methods for each option? Most of them might never be used. Maybe we can just have some `.with(ConfigParameter, Value)` method?
* Should this be moved into the `USerOperatorConfig` class to have private access to some of the fields without needing to expose them through public methods?